extension.js

```diff
@@ -311,6 +320,38 @@ function enable() {
     });
 }
 
+function add_prime_menu_item() {
+    origin = AppDisplay.AppIconMenu.prototype._redisplay;
+
+    AppDisplay.AppIconMenu.prototype._redisplay = function() {
+        let ret = origin.apply(this, arguments);
+
+        let app_info = this._source.app.get_app_info();
+
+        let show_item =
+            !this._source.app.is_window_backed() &&
+            !app_info.list_actions().includes('activate-discrete-gpu');
+
+        if (show_item) {
+            this._appendSeparator();
+
+            let item = this._appendMenuItem(_("Launch using Dedicated Graphics Card"));
+            item.connect('activate', Lang.bind(this, function() {
+                this._source.animateLaunch();
+
+                let prefix = "env __NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia __VK_LAYER_NV_optimus=NVIDIA_only ";
+                let cmdline = prefix + app_info.get_executable();
+
+                Util.trySpawn(cmdline.split(" "));
```

Comment:
I strongly recommend doing this instead:

```
Util.trySpawn([
    "env",
    "__NV_PRIME_RENDER_OFFLOAD=1",
    "__GLX_VENDOR_LIBRARY_NAME=nvidia",
    "__VK_LAYER_NV_optimus=NVIDIA_only",
    app_info.get_executable()
]);

That way if app_info.get_executable() for whatever reason has spaces, this will still work